src/Service/DocumentUploader.php

```diff
@@ -73,11 +73,15 @@ public function createDocumentFromUploadedFile(UploadedFile $file, string $docum
         $tempDir = sys_get_temp_dir();
         $file = $file->move($tempDir, $file->getClientOriginalName());
 
+        $tempPath = $file->getRealPath();
+
         $document = $this->createDocumentFromPath($file->getRealPath(), $documentName, $documentType);
 
         // Explicitly indicate that this is created manually, i.e. display original file name on document index page.
         $document->setIsCreatedManually(true);
 
+        unlink($tempPath);
+
```

Comment:
Consider using `try … finally` here (cf. https://www.php.net/manual/en/language.exceptions.php#language.exceptions.finally) to make sure that the file is actually deleted even if something goes wrong when creating the document, i.e.

```php
try {
  $tempPath = …
  …
} finally {
  unlink($tempPath);
}
```

Reply:
Good idea. Updated in b9c8983570afa1efd12f68ce397d899315fb2fe1.